File: Javascript/섹션2. 1-2차원 배열 탐색/4. 점수 계산 (하)/AA.js
function solution(num, arr){
    let sum=0
    let cnt=0

    let score=0
    for(let i=0;i<num;i++){
        score=arr[i]
        if(score==1){
            cnt++
            sum+=cnt
        }
        else if(score==0){
            cnt=0
        }
    }

    return sum
}
let arr=[1, 0, 1, 1, 1, 0, 0, 1, 1, 0];
console.log(solution(10, arr))

File: Python/섹션 3 - (진행중) 탐색&시뮬레이션(string,1차원,2차원 리스트 탐색)/5. 수들의 합 (상)/AA.py
import sys
#sys.stdin=open('input.txt', 'rt')

n,m=map(int,input().split())
arr=list(map(int,input().split()))

cnt=0
sum=arr[0] # 첫 원소는 가지고 시작
lp=0 # right pointer
rp=1 # left pointer
while True:
    if sum<m:
        if rp<n:
            sum+=arr[rp]
            rp+=1
        else:
            break
    elif sum==m:
        if lp<rp: # 에러 방지
            cnt+=1
            sum-=arr[lp]
            lp+=1
    else: # sum>m
        if lp<rp: # 에러 방지
            sum-=arr[lp]
            lp+=1




print(cnt)